### PR TITLE
Added method for managers to update a user's email

### PR DESCRIPTION
**Pull request by raul-fuentes**
References
==========
https://github.com/consul/consul/issues/2499

Objectives
==========
Added method for a manager user to change emails of confirmed users

Visual Changes (if any)
=======================

peek 2018-03-14 16-24

Notes
=====================
none

### CODE REVIEW

File: app/views/management/users/_erase_user_account.html.erb [hunk 1]
```diff
@@ -1,9 +1,11 @@
-<%= link_to t("management.users.erase_account_link"), "#", class: "button hollow alert js-toggle-link", data: { "toggle-selector" => "#erase-account-form" } %>
+<%= link_to t("management.users.erase_account_link"), "#", class: "button hollow alert js-toggle-link edit-account-form", data: { "toggle-selector" => "#erase-account-form" },
```
Comment (MariaCheca) [inline app/views/management/users/_erase_user_account.html.erb]: I would indent this element more like [this one here](https://github.com/AyuntamientoMadrid/consul/blob/master/app/views/budgets/index.html.erb#L41) to make it more readable.

File: config/locales/en/management.yml [hunk 2]
```diff
@@ -121,6 +121,8 @@ en:
       erase_account_confirm: Are you sure you want to erase the account? This action can not be undone
       erase_warning: This action can not be undone. Please make sure you want to erase this account.
       erase_submit: Delete account
+      edit_account: Edit account
```
Comment (MariaCheca) [inline config/locales/en/management.yml]: As the email is the only field we're changing here...I would go for an "Edit email" and "Update email"

File: config/locales/es/management.yml [hunk 3]
```diff
@@ -121,6 +121,8 @@ es:
       erase_account_confirm: '¿Seguro que quieres borrar a este usuario? Esta acción no se puede deshacer'
       erase_warning: Esta acción no se puede deshacer. Por favor asegúrese de que quiere eliminar esta cuenta.
       erase_submit: Borrar cuenta
+      edit_account: Editar cuenta
```
Comment (MariaCheca) [inline config/locales/es/management.yml]: Same as the comment above.

File: app/views/management/account/show.html.erb [hunk 4]
```diff
@@ -1,3 +1,4 @@
 <h2><%= t("management.account.show.title") %></h2>
 
 <%= render 'management/users/erase_user_account' %>
+<%= render 'management/users/edit_user_account' %>
```
[the remark this answers is not in the document]
Comment (raul-fuentes) [reply app/views/management/account/show.html.erb]: In the notes section warns about users without email, in that case the form will appear empty.
Maybe there is something that I don't understand, but any user can have an email even if they don't have one at the moment, isn't it?

[the remark this answers is not in the document]
Comment (bertocq) [reply app/views/management/account/show.html.erb]: I think you're right @raul-fuentes. The `edit_user_account` form should always be displayed as scenarios are:

**Comment by bertocq** [comment]
@raul-fuentes after changing the "Edit account" texts that maria requested, you're missing some changes in the specs expectations as well https://travis-ci.org/AyuntamientoMadrid/consul/jobs/355308112 at `spec/features/management/account_spec.rb`

File: app/views/management/account/show.html.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (raul-fuentes) [reply app/views/management/account/show.html.erb]: That's the actual behaviour, should I remark it some way?

[the remark this answers is not in the document]
Comment (bertocq) [reply app/views/management/account/show.html.erb]: No it's good 👍 I just wrote it as "I think.." because I needed to double check it afterwards

**Comment by bertocq** [comment]
@raul-fuentes it's always a good idea to comment on the PR so we can notice requested changes has been applied 👍 we don't cycle through all pending changes PR's refreshing to see if there are changes.

**Comment by raul-fuentes** [comment]
Thanks for the advise, I 'll do it from now on

**Comment by bertocq** [comment]
I'm sorry to contradict decabeza's mockups, but after reviewing https://github.com/consul/consul/issues/2500 as well, this is going to get crazy if we keep adding more and more options as ajax forms that appear and dissapear as a toggle

kapture 2018-03-21 at 21 06 22

We're talking about a section that is used by Public Workers only:
- Not end users, no need to fancy them with animations
- Probably on legacy systems (think Windows XP SP1 with IE6 or IE7 if we're lucky)
- Used to hardcore old school apps with menus and stuff written in Java, not in JS.

So with @decabeza 's agreement I'm proposing to:

1- Create a submenu under "Edit user account" menu item with "Edit Email" title. That menu will display the email change form (✅ working as expected) and nothing more
2- Create a second submenu under "Edit user account" with title "Delete User" that will have that pre-existing "Delete User" button that is toggled to the email form
3- Removing the "Edit email" button from the to right corner
4- Keep the new red "Change user" link in the left bottom corner 👍 I think at least the red color improves the usability as its a "destructive" action.

Also with this we can avoid those `render_fails` and toggle usages that really doesn't add value to the feature, only make the code harder to follow and maintain in the future if we keep adding stuff to this panel

**Comment by raul-fuentes** [comment]
I have completed the requested changes and so do @iagirre, should we merge our PR's into one? We think that doing so we could easily merge the interface

**Comment by bertocq** [comment]
Great! Those changes in the menu look great 👍 But I still have some concerns:

## [1] User experience still is not good enough
After the manager sets the new email, its true on localhost we are redirected to the email the user would see... but the Manager in Production will not, he'll see:

Instead he will see the following:
kapture 2018-03-22 at 19 57 06

The old email is still present, and there's no clue about what just happened, just a "Cambios guardados" that could be good enough if at least the manager or the user had our knowledge (that an email has been sent and everything else) but they don't, and assuming they know might get us some tickets "email change not working properly". Again, this an interface for public workers helping really-low-technical-knowledge users.

Think about showing under "Cuenta de usuario" a message that clearly explains what just happened and next steps, like you greatly did with the "Borrar cuenta" submenu.

`Se ha enviado un mensaje de correo electrónico a la nueva dirección de email introducida. Será necesario usar un enlace que aparece en ese mensaje para que el cambio de email se complete con éxito`

`An email message has been sent on to the introduced email address. It will be necessary to follow a link on that email message for the email address change to be completed effectively.`

## [2] Commit history needs rebase
After changes for [1] are done, I would recommend to rebase all commits as the final result will be hard to follow for a person in the future trying to figure out either what new feature we added or why a change in a certain file was done. As a personal tip, I sometimes just go:
```
git reset master
git add..
git commit..
```
as rebasing multiple commits can be far tedious than just creating them from scratch

**Comment by bertocq** [comment]
As you can see https://github.com/AyuntamientoMadrid/consul/pull/1370 was merged and some conflicts appeared (nothing serious as far as I have seen). Your PR has a fix for an spec that was failing in itziar's 👍, and the "Change user" red link 👌 . But it would be nice it you could merge from his changes the "Dropdown" effect that the her PR added:

kapture 2018-03-22 at 23 00 43

**Comment by raul-fuentes** [comment]
@bertocq all the requested changes are done. Just one thing.
I changed the flash message after sent the new mail conformation request this way

screenshot from 2018-03-23 10-47-38

Its correct or is too long and maybe its better to show it in a green warn square like the warning before deleting an account?

**Comment by bertocq** [comment]
@raul-fuentes yes I was thinking about an square below the "Cuenta de usuario" title.

> Think about showing under "Cuenta de usuario" a message that clearly explains what just happened and next steps (like you greatly did with the "Borrar cuenta" submenu 👏)

**Comment by raul-fuentes** [comment]
@bertocq here are the new changes.

screenshot from 2018-03-23 15-53-05

**Comment by bertocq** [comment]
Great @raul-fuentes ! Only left the merge conflicts

**Comment by raul-fuentes** [comment]
@bertocq, also merged the conflicts
this are the results:

screenshot from 2018-03-23 16-26-40

**Comment by bertocq** [comment]
There are some spec/i18n_spec.rb and spec/features/management/users_spec.rb test failures raul :(

**Comment by raul-fuentes** [comment]
Hello @bertocq  I just saw Travis a moment ago!
I have already made the necessary changes so that the tests work properly.

**Comment by raul-fuentes** [comment]
PR updated!
